.github/workflows/main.yml

```diff
@@ -0,0 +1,94 @@
+name: Playwright Tests
+on:
+  pull_request:
+    branches:
+      - develop
+      - master
+      - release-*
+jobs:
+  
+  linux:
+    name: ${{ matrix.os }} (${{ matrix.browser }})
+    strategy:
+      fail-fast: false
+      matrix:
+        browser: [chromium, firefox]
+        os: [ubuntu-18.04]
+        nodeversion: ['16']
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.nodeversion }}
+    - name: Setup custom host for sdkapp.example.com
+      run: echo "127.0.0.1 sdkapp.example.com sdks.mytestrun.com sdkapp.petrov.ca auth.example.com openig.example.com api.example.com user.example.com rest.example.com bank.example.com openam.example.com spa.example.com sdk.mytestrun.com sdks.mytestrun.com forgerock-sdk.local" | sudo tee -a /etc/hosts
+    - name: Install NPM
+      run: npm i -g npm@8
+    - name: Install Dependencies
+      run: npm ci
+      env:
+        DEBUG: pw:install
+        PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
+    - name: Build SDK
+      run: npm run build
+    - name: Install Playwright
+      run: npx playwright install --with-deps
+    - name: Run e2e
+      run: npm run test:e2e --project=${{ matrix.browser }}
+      env:
+        NODE_TLS_REJECT_UNAUTHORIZED: 0  
+  osx:
+      name: ${{ matrix.os }} (${{ matrix.browser }})
+      strategy:
+        fail-fast: false
+        matrix:
+          browser: [chromium, firefox, webkit]
+          os: [macos-11.0]
+          nodeversion: ['16']
+      runs-on: ${{ matrix.os }}
+      steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.nodeversion }}
+      - name: Setup custom host for sdkapp.example.com
+        run: echo "127.0.0.1 sdkapp.example.com sdks.mytestrun.com sdkapp.petrov.ca auth.example.com openig.example.com api.example.com user.example.com rest.example.com bank.example.com openam.example.com spa.example.com sdk.mytestrun.com sdks.mytestrun.com forgerock-sdk.local" | sudo tee -a /etc/hosts      
+      - name: Install NPM
+        run: npm i -g npm@8
+      - name: Install Dependencies
+        run: npm ci
+        env:
+          DEBUG: pw:install
+          # PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
+      # - run: npx playwright install --with-deps ${{ matrix.browser }} chromium
```

Comment:
Because I am not skipping browser downloads here, I don't need to install playwright on the side.

I will align these efforts across all the test scenarios.